sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -216,19 +216,23 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
         c
       }
 
-    case DropTable(ResolvedV1TableIdentifier(ident), ifExists, purge) =>
+    case DropTable(ResolvedV1Identifier(ident), ifExists, purge) =>
       DropTableCommand(ident, ifExists, isView = false, purge = purge)
 
-    case DropTable(_: ResolvedPersistentView, ifExists, purge) =>
-      throw QueryCompilationErrors.cannotDropViewWithDropTableError
-
     // v1 DROP TABLE supports temp view.
-    case DropTable(ResolvedTempView(ident, _), ifExists, purge) =>
-      DropTableCommand(ident.asTableIdentifier, ifExists, isView = false, purge = purge)
+    case DropTable(ResolvedIdentifier(FakeSystemCatalog, ident), _, _) =>
+      DropTempViewCommand(ident)
 
-    case DropView(ResolvedViewIdentifier(ident), ifExists) =>
+    case DropView(ResolvedV1Identifier(ident), ifExists) =>
       DropTableCommand(ident, ifExists, isView = true, purge = false)
 
+    case DropView(r @ ResolvedIdentifier(catalog, ident), _) =>
+      if (catalog == FakeSystemCatalog) {
+        DropTempViewCommand(ident)
+      } else {
+        throw QueryCompilationErrors.catalogOperationNotSupported(catalog, "views")
+      }
```

Comment:
nit: `r` not used? And if you make `FakeSystemCatalog` a `case object` it can participate directly in matching here:
```scala
case DropView(ResolvedIdentifier(FakeSystemCatalog, ident), _) => 
  DropTempViewCommand(ident)
case DropView(ResolvedIdentifier(catalog, _), _) => 
  throw ...
```